app/src/lib/menu-update.ts

```diff
@@ -155,8 +155,15 @@ function getMenuState(state: IAppState): Map<MenuIDs, IMenuItemState> {
     menuStateBuilder.setEnabled('pull', hasPublishedBranch && !networkActionInProgress)
     menuStateBuilder.setEnabled('create-branch', !tipStateIsUnknown)
   } else {
-    for (const id of repositoryScopedIDs) {
-      menuStateBuilder.disable(id)
+    if (selectedState && selectedState.type === SelectionType.MissingRepository) {
+      menuStateBuilder.enable('view-repository-on-github')
```

Comment:
If the repository isn't a GitHub repository (i.e., doesn't have a `gitHubRepository`), this should be disabled.